tests/py3/test_lookup.py

```diff
@@ -365,3 +365,34 @@ def test_required_args(target, args, kwargs):
     # Mostly checking that `self` (and only self) is correctly excluded
     st.builds(target, *map(st.just, args),
               **{k: st.just(v) for k, v in kwargs.items()}).example()
+
+
+@pytest.mark.parametrize('thing', [
+    typing.Optional, typing.List, getattr(typing, 'Type', typing.Set)
+])  # check Type if it's available, otherwise Set is redundant but harmless
+def test_cannot_resolve_bare_forward_reference(thing):
+    with pytest.raises(InvalidArgument):
+        t = thing['int']
+        if type(getattr(t, '__args__', [None])[0]) != typing._ForwardRef:
+            assert sys.version_info[:2] == (3, 5)
+            pytest.xfail('python 3.5 typing module is really weird')
+        st.from_type(t).example()
+
+
+class Tree:
```

Comment:
Would it be worth having a second recursive example that uses `NamedTuple`, just on the off-chance that it's doing something strange because typing module?

Reply:
Won't work - the forward references in annotations on a class `__init__` are resolved by `typing.get_type_hints` , but it doesn't do that for recursive `NamedTuple`.